### ISSUE TITLE
SA1413 doesn't accurately describe it's failure condition

### ISSUE THREAD

**Issue by yaakov-h** [DotNetAnalyzers/StyleCopAnalyzers#3433]
Consider the following code:

```csharp
return nodeIn.OperatorKind switch
{
	BinaryOperatorKind.Equal => (x) => (bool)OpenTypeMethods.Equal(OpenTypeMethods.GetValue(x, columnName), value),
	BinaryOperatorKind.NotEqual => (x) => (bool)OpenTypeMethods.NotEqual(OpenTypeMethods.GetValue(x, columnName), value),
	BinaryOperatorKind.GreaterThan => (x) => (bool)OpenTypeMethods.GreaterThan(OpenTypeMethods.GetValue(x, columnName), value),
	BinaryOperatorKind.GreaterThanOrEqual => (x) => (bool)OpenTypeMethods.GreaterThanOrEqual(OpenTypeMethods.GetValue(x, columnName), value),
	BinaryOperatorKind.LessThan => (x) => (bool)OpenTypeMethods.LessThan(OpenTypeMethods.GetValue(x, columnName), value),
	BinaryOperatorKind.LessThanOrEqual => (x) => (bool)OpenTypeMethods.LessThanOrEqual(OpenTypeMethods.GetValue(x, columnName), value),
	_ => throw new InvalidOperationException($"{nodeIn.OperatorKind} is not a supported binary operator")
};
```

This triggers the following diagnostic in v1.2.0-beta.376:

> SA1413: Use trailing comma in multi-line initializers

This is not a multi-line initializer. This is a switch expression.

**Comment by sharwell** [comment]
💭 Maintaining multiple messages here seems like maybe more effort than it's worth. Maybe there's a short message that can apply equally?

**Comment by yaakov-h** [comment]
I can't think of anything succinct that does double-duty well, I was hoping you could 😂

**Comment by ILikePizza555** [comment]
I was going to open a new issue about SA1413 triggering on switch expressions for this reason, but the responses in this issue make me believe that this is intended behavior?

I don't understand why this is the case, generally for switch expressions, the last line of the expression is always a default case, nothing will come after it, it will only have lines added before it. Not having a comma is pretty intentional here.
